recipes/cose-c/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class CoseCStackConan(ConanFile):
+    name = "cose-c"
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/cose-wg/COSE-C"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = """Implementation of COSE in C using cn-cbor and openssl"""
+    topics = ("cbor")
+    exports_sources = ['CMakeLists.txt']
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_mbedtls": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_mbedtls": False
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def requirements(self):
+        self.requires("cn-cbor/1.0.0")
+
+        if self.options.use_mbedtls:
+            self.requires("mbedtls/2.16.3-gpl")
+        else:
+            self.requires("openssl/1.1.1d")
+
+    def configure(self):
+        if self.settings.os in ["Windows", "Macos", "Android"]:
+            raise ConanInvalidConfiguration("cose-c does not support your OS at the moment")
```

Comment:
The upstream informs the error, but it's related to the compiler, not by OS. I think is better following the same behavior: https://github.com/cose-wg/COSE-C/blob/master/CMakeLists.txt#L112